From 9b6df01de8944114d5563a434df143be1ed2dad1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Kundr=C3=A1t?= <jan.kundrat@cesnet.cz>
Date: Thu, 19 Oct 2017 11:45:23 +0200
Subject: [PATCH 5/5] clearfog: Move all extra bootargs to a special variable

I have two reasons for this, and both of them center around making it
possible to perform changes to the boot process.

It is quite dangerous to attempt to re-flash the entire card, including
the bootloader, on the fly. Moving as much bits as possible to the
environment makes it a bit easier to introduce important changes such
as, for example, volatile rootfs, later on.

The second one is for testing -- it's much easier to just rewrite one
reasonable variable. Changes no longer require struggling with overly
long lines (bootcmd), at least.
---
 include/configs/clearfog.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/clearfog.h b/include/configs/clearfog.h
index d251278885..04faf5c49f 100644
--- a/include/configs/clearfog.h
+++ b/include/configs/clearfog.h
@@ -178,6 +178,7 @@
 	RELOCATION_LIMITS_ENV_SETTINGS \
 	LOAD_ADDRESS_ENV_SETTINGS \
 	"console=ttyS0,115200\0" \
+	"extra_bootargs=panic=10 oops=panic\0" \
 	"bootcmd=test -n \"${BOOT_A_LEFT}\" || setenv BOOT_A_LEFT 3;" \
 	"test -n \"${BOOT_B_LEFT}\" || setenv BOOT_B_LEFT 3;" \
 	"setenv rauc_part;" \
@@ -209,7 +210,7 @@
 	"  saveenv;" \
 	"  reset;" \
 	"fi;" \
-	"setenv bootargs root=/dev/mmcblk0p${rauc_part} rauc.slot=${rauc_slot} czechlight=${czechlight} panic=10 oops=panic;" \
+	"setenv bootargs root=/dev/mmcblk0p${rauc_part} rauc.slot=${rauc_slot} czechlight=${czechlight} ${extra_bootargs};" \
 	"load mmc 0:${rauc_part} ${fdt_addr_r} /boot/armada-388-clearfog-base.dtb || reset;" \
 	"load mmc 0:${rauc_part} ${kernel_addr_r} /boot/zImage || reset;" \
 	"bootz ${kernel_addr_r} - ${fdt_addr_r} || reset\0"
-- 
2.14.1

